core/src/main/scala/chisel3/internal/Builder.scala

```diff
@@ -903,6 +903,12 @@ private[chisel3] object Builder extends LazyLogging {
         case Clone(m: experimental.hierarchy.ModuleClone[_]) => namer(m.getPorts, prefix)
         case _ =>
       }
+    case (d: Data) =>
+      // Views are often returned in lieu of the original, so name the original (as appropriate)
+      // If a view but not identity, return the view and name it since it shows up in .toString and error messages
+      // TODO recurse on targets of non-identity views, perhaps with additional prefix from the view
```

Comment:
```suggestion
      // Views are often returned in lieu of the target, so name the target (as appropriate).
      // If a view but not identity, return the view and name it since it shows up in .toString and error messages.
      // TODO recurse on targets of non-identity views, perhaps with additional prefix from the view.
```